### ISSUE TITLE
Compile vcpkg with PCH with non-MSVC compilers

### ISSUE THREAD

**Issue by linquize** [microsoft/vcpkg#12388]
Currently, vcpkg only enables PCH for MSVC.
Compile vcpkg with PCH with non-MSVC compilers

New cmake function `target_precompile_headers` is introduced in CMake 3.16, which can handle both MSVC and non-MSVC compilers.

**Comment by linquize** [comment]
https://github.com/microsoft/vcpkg/blob/895678db8f8bfc3c1988a118b3bb5a13ea4d5248/toolsrc/CMakeLists.txt#L100

Simply replace this line with `target_precompile_headers`

**Comment by JackBoosY** [comment]
Is this needed?

**Comment by JackBoosY** [comment]
This issue can be closed since vcpkg provides pre-build binary now.

**Comment by niyue** [comment]
@JackBoosY could you please give more clue on how users can use the `pre-build binary` to address this issue? I thought pre compiled headers will make using those header only library faster (like `Boost.Asio`), but for header only library, vcpkg will only copy these header files into the installed directory and do nothing else. How does the `pre-build` binary can help in this case? Thanks.

**Comment by JackBoosY** [comment]
@niyue We may talk in the different thread,
According to the code provided by the author, this is obviously talking about the pch problem when building vcpkg, not the problem of using pch in the port.

**Comment by niyue** [comment]
@JackBoosY thanks for pointing it out. I am looking for solution about how I can use precompiled header in vcpkg by searching `target_precompile_headers vcpkg` and this was the first result in Google so I subscribed to the thread long ago, and apparently I misunderstood this issue. 

And just in case other people searching the same keywords come here, I created another vcpkg discussion thread [here](https://github.com/microsoft/vcpkg/discussions/22849) to discuss how to use pch for vcpkg port.
